extensions/vault/deployment/src/main/java/io/quarkus/vault/VaultProcessor.java

```diff
@@ -32,13 +35,18 @@
     void build(
             BuildProducer<ReflectiveClassBuildItem> reflectiveClasses,
             BuildProducer<FeatureBuildItem> feature,
-            CombinedIndexBuildItem combinedIndexBuildItem,
             SslNativeConfigBuildItem sslNativeConfig,
-            BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
+            BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) throws Exception {
 
         feature.produce(new FeatureBuildItem(Feature.VAULT));
 
-        final String[] modelClasses = combinedIndexBuildItem.getIndex()
+        // Manually index the runtime module because we need to find all model classes in it
```

Comment:
This is needed because without `beans.xml` in the runtime module, the model classes no longer appear in the `CombinedIndexBuildItem`.
This might be a crazy (and slow) way to find all those classes, so suggestions welcome how to do it better. But I couldn't think of a different way than listing all the classes manually, which would be very ugly, because there's like 100 of them :((

Reply:
this sound reasonable. I do not know personally a better way, but I am not a specialist of that area. may be @mkouba would know?

Reply:
I personally think it's fine